pkg/dataplane/conntrack.go

```diff
@@ -0,0 +1,114 @@
+package dataplane
+
+import (
+	"crypto/sha256"
+	"encoding/binary"
+	"fmt"
+
+	"github.com/vishvananda/netlink"
+	"golang.org/x/sys/unix"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/kube-network-policies/pkg/network"
+)
+
+var (
+	mapIPFamilyToString = map[uint8]v1.IPFamily{
+		unix.AF_INET:  v1.IPv4Protocol,
+		unix.AF_INET6: v1.IPv6Protocol,
+	}
+	mapProtocolToString = map[uint8]v1.Protocol{
+		unix.IPPROTO_TCP:  v1.ProtocolTCP,
+		unix.IPPROTO_UDP:  v1.ProtocolUDP,
+		unix.IPPROTO_SCTP: v1.ProtocolSCTP,
+	}
+)
+
+func PacketFromFlow(flow *netlink.ConntrackFlow) *network.Packet {
+	if flow == nil {
+		return nil
+	}
+	packet := network.Packet{
+		SrcIP:   flow.Forward.SrcIP,
+		DstIP:   flow.Reverse.SrcIP,
+		SrcPort: int(flow.Forward.SrcPort),
+		DstPort: int(flow.Reverse.SrcPort),
+	}
+
+	if family, ok := mapIPFamilyToString[flow.FamilyType]; ok {
+		packet.Family = family
+	} else {
+		klog.V(4).Info("Unknown IP family", "family", flow.FamilyType, "flow", flow)
+		if flow.Forward.SrcIP.To4() != nil {
+			packet.Family = v1.IPv4Protocol
+		} else {
+			packet.Family = v1.IPv6Protocol
```

Comment:
Why would `flow.FamilyType` not be set correctly if the flow actually was an IPv4 or IPv6 flow? This seems like "can't happen". I would just do

`packet.Family = mapIPFamilyToString[flow.FamilyType]`

Reply:
there are more protocols and I do not know if is possible to get other protocol through the conntrack , but as you say if is not ipv4 or ipv6 the IP field will not exist